src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1352 @@ ActorInit Boss_05_InitVars = {
+        (limbIndex == BIO_DEKU_BABA_LILY_PAD_LIMB_LEAF)) {
+        *dList = NULL;
+    }
+
+    return false;
+}
+
+void Boss05_TransformLimbDraw_FallingHeadLilyPad(PlayState* play, s32 limbIndex, Actor* thisx) {
+    Boss05* this = THIS;
+
+    if ((limbIndex >= BIO_DEKU_BABA_LILY_PAD_LIMB_MIDDLE_STEM) &&
+        (limbIndex <= BIO_DEKU_BABA_LILY_PAD_LIMB_RIGHT_LOWER_ARM)) {
+        Matrix_Scale(this->fallingHeadLilyPadLimbScale, this->fallingHeadLilyPadLimbScale,
+                     this->fallingHeadLilyPadLimbScale, MTXMODE_APPLY);
+    }
+}
+
+static BioDekuBabaHeadLimb sFragmentIndexToLimbIndex[BIO_BABA_TYPE_MAX - BIO_BABA_TYPE_FRAGMENT_BASE] = {
+    BIO_DEKU_BABA_HEAD_LIMB_LOWER_JAW,            // BIO_BABA_TYPE_FRAGMENT_LOWER_JAW
+    BIO_DEKU_BABA_HEAD_LIMB_UPPER_JAW,            // BIO_BABA_TYPE_FRAGMENT_UPPER_JAW
+    BIO_DEKU_BABA_HEAD_LIMB_BODY,                 // BIO_BABA_TYPE_FRAGMENT_BODY
+    BIO_DEKU_BABA_HEAD_LIMB_BACK_UPPER_LEG,       // BIO_BABA_TYPE_FRAGMENT_BACK_UPPER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_BACK_LOWER_LEG,       // BIO_BABA_TYPE_FRAGMENT_BACK_LOWER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_LEFT_UPPER_LEG,       // BIO_BABA_TYPE_FRAGMENT_LEFT_UPPER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_LEFT_LOWER_LEG,       // BIO_BABA_TYPE_FRAGMENT_LEFT_LOWER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_LEFT_LOWER_EYESTALK,  // BIO_BABA_TYPE_FRAGMENT_LEFT_LOWER_EYESTALK
+    BIO_DEKU_BABA_HEAD_LIMB_LEFT_UPPER_EYESTALK,  // BIO_BABA_TYPE_FRAGMENT_LEFT_UPPER_EYESTALK
+    BIO_DEKU_BABA_HEAD_LIMB_RIGHT_LOWER_EYESTALK, // BIO_BABA_TYPE_FRAGMENT_RIGHT_LOWER_EYESTALK
+    BIO_DEKU_BABA_HEAD_LIMB_RIGHT_UPPER_EYESTALK, // BIO_BABA_TYPE_FRAGMENT_RIGHT_UPPER_EYESTALK
+    BIO_DEKU_BABA_HEAD_LIMB_RIGHT_UPPER_LEG,      // BIO_BABA_TYPE_FRAGMENT_RIGHT_UPPER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_RIGHT_LOWER_LEG,      // BIO_BABA_TYPE_FRAGMENT_RIGHT_LOWER_LEG
+    BIO_DEKU_BABA_HEAD_LIMB_LEAVES,               // BIO_BABA_TYPE_FRAGMENT_LEAVES
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_05/func_809F0474.s")
+s32 Boss05_OverrideLimbDraw_Fragment(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                     Actor* thisx) {
+    Boss05* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_05/func_809F04C0.s")
+    if (limbIndex != sFragmentIndexToLimbIndex[BIO_BABA_GET_FRAGMENT_INDEX(&this->dyna.actor)]) {
+        *dList = NULL;
+    } else if (this->fragmentState >= BIO_BABA_FRAGMENT_STATE_ABOVE_WATER) {
+        rot->x += this->frameCounter * 0x3000;
+        rot->y += this->frameCounter * 0x1A00;
+        rot->z += this->frameCounter * 0x2000;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_05/func_809F0538.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_05/func_809F0590.s")
+void Boss05_PostLimbDraw_Fragment(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
```

Comment:
I feel like Fragment should come before PostLimbDraw in the name, so that the actor type comes first. Same for the other Override/Post/TransformLimbDraws

Reply:
The reason I didn't do this is because this isn't always an actor type. For example, `Boss05_TransformLimbDraw_Head`. There isn't a "Head" actor type, this is just used for all heads, regardless of what type of actor they are.

Now, if you'd still prefer it before PostLimbDraw, that's fine. I just want to make sure it's clear this is *not* an actor type; putting it *after* the PostLimbDraw in the name was a deliberate thing I did to show that it *wasn't* tied to an actor type, unlike other functions like `Boss05_LilyPad_Move` for example.

Reply:
I thought about it some more and checked to see if anyone put anything after `PostLimbDraw` like I'm doing here (nobody is). While I do feel like making it clear that these aren't necessarily actor types is still valuable, it doesn't feel valuable to go against the convention set by literally every other actor in the codebase, so I swapped it.